app/src/main/java/com/example/parkingsystemkotlin/utils/ReservationChecker.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.parkingsystemkotlin.utils
+
+import com.example.parkingsystemkotlin.database.ParkingDatabase
+import com.example.parkingsystemkotlin.entity.Reservation
+
+class ReservationChecker(private val dataBase: ParkingDatabase = ParkingDatabase) {
+
+    fun checkReservation(reservation: Reservation?): EnumReservationVerify {
+        if (reservation?.isStarDateInitialized == false) return EnumReservationVerify.MISSING_DATESTART
+        if (reservation?.isEndDateInitialized == false) return EnumReservationVerify.MISSING_DATEEND
+        if (reservation?.parkingLot == Constant.NOT_VALID) return EnumReservationVerify.MISSING_LOT
+        if (reservation?.userCode == Constant.NOT_VALID) return EnumReservationVerify.MISSING_CODE
+        if (!reservationOK(reservation)) return EnumReservationVerify.RESERVATION_OVERLAP
+        return EnumReservationVerify.COMPROBATION_OK
+    }
+
+    private fun reservationOK(reservation: Reservation?): Boolean {
+        val lot: Int? = reservation?.parkingLot
+        if (dataBase.getHashMapReservation().containsKey(lot)) {
+            val reservationsOnLot: List<Reservation>? = dataBase.getHashMapReservation()[lot]
+            if (!reservationsOnLot.isNullOrEmpty()) {
+                for (i in reservationsOnLot) {
```

Comment:
please use more descriptive names

Reply:
you can use Collection method to improve this loop

Reply:
Done